Hardware/Sources/Hardware/CardReaderService.swift

```diff
@@ -0,0 +1,58 @@
+import Combine
+
+/// Abstracts the integration with a Card Reader
+public protocol CardReaderService {
+
+    // MARK: - Queries
+    /// The publisher that emits the list of discovered readers whenever the service discovers a new reader.
+    var discoveredReaders: AnyPublisher<[CardReader], Never> { get }
+
+    /// The Publisher that emits the connected readers
+    var connectedReaders: AnyPublisher<[CardReader], Never> { get }
+
+    /// The Publisher that emits the service status
+    var serviceStatus: AnyPublisher<CardReaderServiceStatus, Never> { get }
+
+    /// The Publisher that emits the service discovery status
+    var discoveryStatus: AnyPublisher<CardReaderServiceDiscoveryStatus, Never> { get }
+
+    /// The Publisher that emits the payment status
+    var paymentStatus: AnyPublisher<PaymentStatus, Never> { get }
+
+    /// The Publisher that emits reader events
+    var readerEvent: AnyPublisher<CardReaderEvent, Never> { get }
+
+    // MARK: - Commands
+
+    /// Starts the service.
+    /// That could imply, for example, that the reader discovery process starts
+    func start()
```

Comment:
In the next iteration of this abstraction, let's have initialization, discovery and connection be things handled separately.